### PR TITLE
Add support for Pod fields in environment variables

### LINKED ISSUE #182: Add support for Pod fields in environment variables

### PR DESCRIPTION
**Pull request by marcinc**
Resolves #182

### CODE REVIEW

File: pkg/kev/converter/kubernetes/transform.go [hunk 1]
```diff
@@ -1151,37 +1151,65 @@ func (k *Kubernetes) configEnvs(projectService ProjectService) ([]v1.EnvVar, err
...
+			paths := []string{
+				"metadata.name", "metadata.namespace", "metadata.labels", "metadata.annotations",
+				"spec.nodeName", "spec.serviceAccountName", "status.hostIP", "status.podIP", "status.podIPs",
+			}
+
+			path := strings.Join(parts[1:], ".")
+
+			sort.Strings(paths)
+			i := sort.SearchStrings(paths, path)
+			if i < len(paths) && paths[i] == path {
```
Comment (ezodude) [inline pkg/kev/converter/kubernetes/transform.go:1196]: I get how you're using `sort.SearchStrings` to find a `path` in `paths`.

Perhaps you can use a simple `Set` to to do the same without having to resort to `sort` and `SearchStrings` ... for e.g. 
```
paths := map[string]bool{
  "metadata.name": true,
 ...
}
...
if paths[path] {
...
}
```

**Review by ezodude — COMMENTED** [review]
Minor tweak

File: pkg/kev/converter/kubernetes/transform.go (hunk 1 above)
Comment (marcinc) [reply pkg/kev/converter/kubernetes/transform.go:1196]: going to turn original string detection in slice into a helper function as it is used in a couple of places now. sort.SearchStrings seem much more elegant than building a map first to find something in it later.

Comment (marcinc) [reply pkg/kev/converter/kubernetes/transform.go:1196]: ^ Done in the follow up PR: #259

**Review by ezodude — APPROVED** [review]
👍

**Pull request merged**